lms/static/scripts/frontend_apps/components/AutoGradingConfigurator.tsx

```diff
@@ -0,0 +1,208 @@
+   */
+  requiredReplies?: number;
+};
+
+type AnnotationsGoalInputGroupProps = {
+  children?: ComponentChildren;
+  gradingType: GradingType;
+  value: number;
+  onChange: (newValue: number) => void;
+};
+
+/**
+ * Controls containing a number input to set the amount of required annotations
+ * or replies
+ */
+function AnnotationsGoalInputGroup({
+  children,
+  gradingType,
+  value,
+  onChange,
+}: AnnotationsGoalInputGroupProps) {
+  const inputId = useId();
+
+  return (
+    <div className="flex gap-2 items-center">
+      <label
+        className="grow flex justify-between items-center"
+        htmlFor={inputId}
+      >
+        {children}
+        <span className="uppercase font-semibold">
+          {gradingType === 'all_or_nothing' ? 'Minimum' : 'Goal'}
+        </span>
+      </label>
+      <Input
+        id={inputId}
+        classes="max-w-14"
+        type="number"
+        required
+        min={1}
+        step={1}
+        value={value}
+        onChange={e => onChange(Number((e.target as HTMLInputElement).value))}
+      />
+    </div>
+  );
+}
+
+export type AutoGradingConfiguratorProps = {
+  autoGradingConfig: AutoGradingConfig;
+  updateAutoGradingConfig: (newConfig: AutoGradingConfig) => void;
+};
+
+/**
+ * Allows instructors to enable auto grading for an assignment, and provide the
+ * configuration to determine how to calculate every student's grade.
+ */
+export default function AutoGradingConfigurator({
+  autoGradingConfig,
+  updateAutoGradingConfig,
+}: AutoGradingConfiguratorProps) {
+  const {
+    active = false,
+    gradingType,
+    activityCalculation,
+    requiredAnnotations,
+    requiredReplies = 1,
+  } = autoGradingConfig;
+  const updatePartialConfig = useCallback(
+    (partialConfig: Partial<AutoGradingConfig>) =>
+      updateAutoGradingConfig({ ...autoGradingConfig, ...partialConfig }),
+    [autoGradingConfig, updateAutoGradingConfig],
+  );
+
+  const gradingTypeId = useId();
+  const activityCalculationId = useId();
+
+  return (
+    <div className="flex flex-col gap-y-3">
+      <Checkbox
+        checked={active}
+        checkedIcon={CheckboxCheckedFilledIcon}
+        onChange={e =>
+          updatePartialConfig({
+            active: (e.target as HTMLInputElement).checked,
+          })
+        }
+      >
+        Enabled automatic participation grading
+      </Checkbox>
+      {active && (
+        <>
+          <div>
+            <p id={gradingTypeId} className="font-semibold mb-1">
+              Grading type
+            </p>
+            <RadioGroup
+              data-testid="grading-type-radio-group"
+              aria-labelledby={gradingTypeId}
+              selected={gradingType}
+              onChange={gradingType => updatePartialConfig({ gradingType })}
+            >
+              <RadioGroup.Radio
+                value="all_or_nothing"
+                subtitle={<small>Must meet minimum requirements.</small>}
```

Comment:
This results in an 11px font which is probably too small. I can't remember if we agreed on a lower limit anywhere, but around 12px feels appropriate to me. In GitHub's UI that's the size of the "Markdown is supported" note below the input box.

Reply:
There's an issue with the font size in the LMS that we need to solve as a whole. I believe the base is set to 13px, which is probably too low.

I suppose this was chosen either for historical reasons, to deal with the limited space in many places or perhaps a combination of both.